dora/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -448,6 +452,11 @@ protected synchronized void initContext(ClientContext ctx,
     mBlockMasterClientPool = new BlockMasterClientPool(mMasterClientContext);
     mBlockWorkerClientPoolMap = new ConcurrentHashMap<>();
     mUriValidationEnabled = ctx.getUriValidationEnabled();
+    try {
+      mMembershipManager = MembershipManager.Factory.create(getClusterConf());
+    } catch (IOException ex) {
+      LOG.error("Failed to set membership manager.", ex);
```

Comment:
if this ex is caught (instead of thrown up), `mMembershipManager` can be null and cause more trouble later. What about not catching this ex?

Reply:
if its null it will fallback to the asking master for worker list in getAllWorkers

Reply:
[ADDRESS IN NEXT PR]